packages/cli/src/metadataGeneration/typeResolver.ts

```diff
@@ -111,13 +113,15 @@ export class TypeResolver {
     if (ts.isTypeLiteralNode(this.typeNode)) {
       const properties = this.typeNode.members.filter(ts.isPropertySignature).reduce<Tsoa.Property[]>((res, propertySignature: ts.PropertySignature) => {
         const type = new TypeResolver(propertySignature.type as ts.TypeNode, this.current, propertySignature, this.context).resolve();
+
+        const def = TypeResolver.getDefault(propertySignature);
         const property: Tsoa.Property = {
           example: this.getNodeExample(propertySignature),
-          default: getJSDocComment(propertySignature, 'default'),
+          default: def,
           description: this.getNodeDescription(propertySignature),
           format: this.getNodeFormat(propertySignature),
           name: (propertySignature.name as ts.Identifier).text,
-          required: !propertySignature.questionToken,
+          required: !propertySignature.questionToken && def === undefined,
```

Comment:
If a type has default value, then it can not be required.
If a string has a number default value (for example, because of a type mapping), then no problem, the not defined value will be replaced with the number, and after that the typeCheck says, it is not the expected type.

Reply:
How does this interact with: https://github.com/lukeautry/tsoa/pull/1454

My approach was to leave that up to  the maping of internal types -> OpenAPI 2/3, while keeping the metadata rather unrefined and applying "intentions" later.

Reply:
Ok, I removed the automatic non-required if there is a default value.